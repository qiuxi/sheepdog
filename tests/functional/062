#!/bin/bash

# Test unaligned dog vdi read/write

. ./common

for i in `seq 0 5`; do
    _start_sheep $i
done

_wait_for_sheep 6

_cluster_format -c 6

_vdi_create test 4095
sleep 1
$DOG vdi list -r test | awk '{$7="MASKED";print $0}'

echo hello | $DOG vdi write test 1 6
$DOG vdi read test 1 6
echo world | $DOG vdi write test 1 6
$DOG vdi read test 1 6

echo h*** | $DOG vdi write test 510 512
echo h!!! | $DOG vdi write test 510 512
$DOG vdi read test 510 5
echo ^^ | $DOG vdi write test 511 2
$DOG vdi read test 510 5
echo xxx | $DOG vdi write test 1020 4
$DOG vdi read test 1020 4
$DOG vdi read test 510 5
$DOG vdi read test 1 6

echo erasure code | $DOG vdi write test 1020 13
echo xxxxxx | $DOG vdi write test  1022 6
$DOG vdi read test 1020 13
