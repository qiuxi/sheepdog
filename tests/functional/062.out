QA output created by 062
using backend plain store
= test 0 4095 0 0 MASKED 7c2b25 6
hello
world
h!!!
h^^!
xxx
h^^!
world
erxxxxxxcode
